packages/react-widgets/src/widgets/PieWidget.js

```diff
@@ -39,33 +38,29 @@ function PieWidget({
   onError,
   wrapperProps
 }) {
+  const dispatch = useDispatch();
+
   const [categoryData, setCategoryData] = useState([]);
   const [selectedCategories, setSelectedCategories] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
 
-  const dispatch = useDispatch();
-  const source = useSelector((state) => selectSourceById(state, dataSource) || {});
-  const viewportFeaturesReady = useSelector((state) => state.carto.viewportFeaturesReady);
-
-  const widgetsLoadingState = useSelector((state) => state.carto.widgetsLoadingState);
```

Comment:
So you're not using widgetsLoadingState anymore right?. In that case we should consider removing it from cartoSlice, the key and related methods

Reply:
Yes, I forgot that.